syntax = "proto3";

package syft.core.node.common;

import "proto/core/common/common_object.proto";
import "proto/core/io/location.proto";
import "proto/core/io/route.proto";

message Client {
  string obj_type = 1;
  syft.core.common.UID id = 2;
  string name = 3;
  repeated syft.core.io.SoloRoute routes = 4;

  bool has_network = 5;
  syft.core.io.SpecificLocation network = 6;

  bool has_domain = 7;
  syft.core.io.SpecificLocation domain = 8;

  bool has_device = 9;
  syft.core.io.SpecificLocation device = 10;

  bool has_vm = 11;
  syft.core.io.SpecificLocation vm = 12;
}
